### PR TITLE
pin-1859: Agreement rejection

### PR DESCRIPTION
**Pull request by galales**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala [hunk 1]
```diff
@@ -190,6 +190,29 @@ final case class AgreementApiServiceImpl(
     )
   }
 
+  override def rejectAgreement(agreementId: String, payload: AgreementRejectionPayload)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerAgreement: ToEntityMarshaller[Agreement]
+  ): Route = authorize(ADMIN_ROLE) {
```
Comment (TonioGela) [inline src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala:197]: Just an admin can refuse an agreement? Is has to be designed like that?

**Review by TonioGela — APPROVED** [review]

File: src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala (hunk 1 above)
Comment (galales) [reply src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala:197]: Right now only admins can change the state of an agreement (activate, suspend, submit, upgrade), so I would say `yes`. But not sure if you had any different use case in mind

Comment (TonioGela) [reply src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala:197]: No, no, I was just asking

File: src/main/resources/interface-specification.yml [hunk 2]
```diff
@@ -168,6 +168,54 @@ paths:
...
+    parameters:
+      - $ref: '#/components/parameters/CorrelationIdHeader'
+      - $ref: '#/components/parameters/IpAddress'
+      - schema:
+          type: string
+          format: uuid
+        name: agreementId
+        in: path
+        required: true
+        description: identifier of the pending agreement to be suspended
```
Comment (beetlecrunch) [inline src/main/resources/interface-specification.yml]: rejected?

**Review by beetlecrunch — APPROVED** [review]

File: src/main/resources/interface-specification.yml (hunk 2 above)
Comment (galales) [reply src/main/resources/interface-specification.yml]: 🙇

**Pull request merged**